llvm/lib/Target/AIE/AIEPostSelectOptimize.cpp

```diff
@@ -266,6 +267,22 @@ bool combineINSERT_SUBREG(MachineBasicBlock &MBB) {
   return Changed;
 }
 
+bool combineVUPS(MachineBasicBlock &MBB) {
+  MachineRegisterInfo &MRI = MBB.getParent()->getRegInfo();
+  const TargetInstrInfo &TII = *MBB.getParent()->getSubtarget().getInstrInfo();
+  auto *AIETII = static_cast<const AIEBaseInstrInfo *>(&TII);
+  bool Changed = false;
+  for (MachineInstr &MI : make_early_inc_range(reverse(MBB))) {
+    if (MI.getOpcode() != TargetOpcode::REG_SEQUENCE ||
+        MI.getNumOperands() != 5)
```

Comment:
Can you fold `MI.getNumOperands() != 5` into the target specific implementation? I think the only requirement for `combineLoHiMI` should be that MI is a REG_SEQUENCE